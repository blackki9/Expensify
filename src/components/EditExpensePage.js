import React from 'react';

const EditExpensePage = (props) =>{
    return (
        <div>
            This is from my edit expense page
        </div>
    );
};

export default EditExpensePage;