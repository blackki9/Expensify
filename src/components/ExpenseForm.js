import React from 'react';
import moment from 'moment';
import 'react-dates/initialize';
import { SingleDatePicker } from 'react-dates';
import 'react-dates/lib/css/_datepicker.css';

export default class ExpenseForm extends React.Component {
    state = {
        description: '',
        note: '',
        amount: '',
        createdAt: moment(),
        calendarFocused: false
    };

    onDescriptionChange = (e) => {
        const description = e.target.value;
        this.setState( () => ({ description }));
    };

    onDateChange = (createdAt) => {
        this.setState(() => ({createdAt}));
    };

    onFocusChanged = ({ focused }) => {
        this.setState( () => ({calendarFocused: focused}));
    };

    onNoteChange = (e) => {
        const note = e.target.value;
        this.setState( () => ( { note } ));
    };

    onAmountChange = (e) => {
        const amount = e.target.value;
        if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
            this.setState( () => ({amount}));
        }
    };

    render() {
        return (
            <div>
                <form>
                    <input
                         type="text"
                         placeholder="Description"
                         autoFocus
                         value={this.state.description}
                         onChange={this.onDescriptionChange}
                    />
                    <input
                         type="text" 
                         value={this.state.amount}
                         onChange={this.onAmountChange}
                         placeholder="Amount" />
                    <SingleDatePicker 
                        date = {this.state.createdAt}
                        onDateChange ={ this.onDateChange}
                        focused = {this.state.calendarFocused}
                        onFocusChange = {this.onFocusChanged}
                        numberOfMonths = {1}
                        isOutsideRange= {(day) => false}
                    />
                    <textarea 
                        value={this.state.note}
                        onChange={this.onNoteChange}
                        placeholder="Add note for your expense"> 

                    </textarea>
                    <button> Add Expense </button>
                </form>

            </div>
        )
    }
}