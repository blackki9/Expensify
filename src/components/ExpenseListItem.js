import React from 'react';

const ExpenseListItem = ({description, amount, createdAt}) => (
    <div>
     {description} {amount} {createdAt}
    </div>
);


export default ExpenseListItem;