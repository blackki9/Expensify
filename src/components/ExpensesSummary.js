import React from 'react';
import numeral from 'numeral';
import {connect} from 'react-redux';

import selectExpenses from '../selectors/expenses';
import totalSumOfExpenses from '../selectors/expenses-total';

export const ExpensesSummary = (props) => {
    const expenseWord = (props.count === 1) ? 'expense' : 'expenses';
    return (
        <div>
           <h1> Viewing {props.count} {expenseWord} totalling {numeral(props.total).format("$0,0.00")} </h1>
        </div>
    );
}

const mapStateToProps = (state) => {
    const visibleExpense = selectExpenses(state.expenses, state.filters);
    return {
        count: visibleExpense.length,
        total: totalSumOfExpenses(visibleExpense)
    }
}

export default connect(mapStateToProps)(ExpensesSummary);