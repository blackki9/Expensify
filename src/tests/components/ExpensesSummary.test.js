import {ExpensesSummary} from '../../components/ExpensesSummary';
import {shallow} from 'enzyme';
import React from 'react';

test('should render summary with 0', () => {
    const wrapper = shallow(<ExpensesSummary />);
    expect(wrapper).toMatchSnapshot();
});

test('should render with one expense', () => {
    const wrapper = shallow(<ExpensesSummary count={1} total={1000412.3} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary count={10} total={9400.3} />);
    expect(wrapper).toMatchSnapshot();
})