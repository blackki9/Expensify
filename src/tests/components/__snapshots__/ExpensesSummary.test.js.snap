// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render summary with 0 1`] = `
<div>
  <h1>
     Viewing 
     
    expenses
     totalling 
    $0.00
     
  </h1>
</div>
`;

exports[`should render with multiple expenses 1`] = `
<div>
  <h1>
     Viewing 
    10
     
    expenses
     totalling 
    $9,400.30
     
  </h1>
</div>
`;

exports[`should render with one expense 1`] = `
<div>
  <h1>
     Viewing 
    1
     
    expense
     totalling 
    $1,000,412.30
     
  </h1>
</div>
`;
